src/main/java/frc/robot/vision/multivisionsources/MultiVisionSources.java

```diff
@@ -0,0 +1,80 @@
+package frc.robot.vision.multivisionsources;
+
+import frc.robot.poseestimator.observations.VisionObservation;
+import frc.robot.subsystems.GBSubsystem;
+import frc.robot.vision.VisionConstants;
+import frc.robot.vision.rawdata.RawVisionData;
+import frc.robot.vision.sources.VisionSource;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class MultiVisionSources<T extends VisionSource<? extends RawVisionData>> extends GBSubsystem {
+
+	private final List<T> visionSources;
+
+	@SafeVarargs
+	public MultiVisionSources(String logPath, T... visionSources) {
+		super(logPath);
+		this.visionSources = List.of(visionSources);
+	}
+
+	public MultiVisionSources(String logPath, List<T> visionSources) {
+		super(logPath);
+		this.visionSources = visionSources;
+	}
+
+	protected List<T> getVisionSources() {
+		return visionSources;
+	}
+
+	public ArrayList<VisionObservation> getAllAvailablePoseData() {
```

Comment:
Raw data

Reply:
Not rawData but VisionObservation. Anyway renamed.